internal/pkg/diagnostics/diagnostics_test.go

```diff
@@ -20,13 +20,98 @@ import (
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
+	"gopkg.in/yaml.v3"
 
 	"github.com/google/pprof/profile"
 
+	agentclient "github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-libs/mapstr"
 	"github.com/elastic/elastic-agent/internal/pkg/agent/application/paths"
+	agentruntime "github.com/elastic/elastic-agent/pkg/component/runtime"
+	"github.com/elastic/elastic-agent/pkg/control/v2/client"
 	"github.com/elastic/elastic-agent/version"
 )
 
+func TestRedactResults(t *testing.T) {
+	exampleConfig := mapstr.M{
+		"root": mapstr.M{
+			"passphrase": "unredacted",
+			"nested1": mapstr.M{
+				"certificate": "unredacted",
+				"nested2": mapstr.M{
+					"passphrase": "unredacted",
+					"password":   "unredacted",
+					"nested3": mapstr.M{
+						"token": "unredacted",
+						"key":   "unredacted",
```

Comment:
Can you use an inline PEM formatted key here (obviously don't use a real one)? We should test that this properly redacts multiline strings.

Reply:
Ah, good idea...